homeassistant/components/network/__init__.py

```diff
@@ -0,0 +1,90 @@
+"""The Network Configuration integration."""
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.components.websocket_api.connection import ActiveConnection
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.loader import bind_hass
+
+from .const import (
+    ATTR_ADAPTERS,
+    ATTR_CONFIGURED_ADAPTERS,
+    DOMAIN,
+    NETWORK_CONFIG_SCHEMA,
+)
+from .models import Adapter
+from .network import Network
+
+ZEROCONF_DOMAIN = "zeroconf"  # cannot import from zeroconf due to circular dep
+_LOGGER = logging.getLogger(__name__)
+
+
+@bind_hass
+async def async_get_adapters(hass: HomeAssistant) -> list[Adapter]:
+    """Get the network adapter configuration."""
+    network: Network = hass.data[DOMAIN]
+    return network.adapters
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up network for Home Assistant."""
+
+    hass.data[DOMAIN] = network = Network(hass)
+    await network.async_setup()
+    await network.async_migrate_from_zeroconf(config.get(ZEROCONF_DOMAIN, {}))
```

Comment:
To save one yield to the event loop, chek if ZEROONF exists.
```suggestion
    if ZEROCONF_DOMAIN in config:
        await network.async_migrate_from_zeroconf(config[ZEROCONF_DOMAIN])
```